server/index.js

```diff
@@ -109,6 +113,54 @@ const getSalt = async event => {
   return { hashes: results }
 }
 
+/**
+ * Allows an authority to report an analysis
+ * @return {Promise<string>} Id of the report
+ */
+const reportAnalysis = async event => {
+  let authorization = event.headers['authorization']
+  if (!authorization) throw new ServerError("Missing header 'authorization'", 400)
+
+  let { date } = JSON.parse(event.body)
+  date = moment(date)
+  if (!date.isValid()) throw new ServerError("Missing or invalid 'date' prop.", 400)
+
+  const { db, client } = await getConnection()
+
+  // TODO: Improve authorization mechanism. Current weak implementation is for hackathon demo purposes
+  const valid = await dbAuthorities.validToken(db, authorization)
+  if(!valid) throw new ServerError("The authority is not valid'", 400)
+
+  const reportId = await dbAnalysisReports.saveReport(db, date)
+    .catch(() => {
+      throw new ServerError("Something went wrong reporting the analysis", 500)
```

Comment:
Perhaps `console.error` the real error so we're not eating errors here?

Reply:
Sure!